examples/nitclk/synchronize-tclk.py

```diff
@@ -0,0 +1,167 @@
+# Link : https://www.ni.com/en-us/support/downloads/drivers/download.ni-fgen.html
+#
+# For instructions on how to use protoc to generate gRPC client interfaces, see our "Creating a gRPC Client" wiki page.
+# Link: https://github.com/ni/grpc-device/wiki/Creating-a-gRPC-Client
+#
+# Running from command line:
+#
+# Server machine's IP address and port number can be passed as separate command line arguments.
+#   > python synchronize-tclk.py <server_address> <port_number>
+# If they are not passed in as command line arguments, then by default the server address will be "localhost:31763"
+
+import grpc
+import sys
+import nifgen_pb2 as nifgen_types
+import nifgen_pb2_grpc as grpc_nifgen
+import nitclk_pb2 as nitclk_types
+import nitclk_pb2_grpc as grpc_nitclk
+import session_pb2 as session_types
+import session_pb2_grpc as grpc_session
+import matplotlib.pyplot as plt
+
+server_address = "localhost"
+server_port = "31763"
+
+# Take resource name . This example doesn't work on simulation, so resource name is required.
+resources = input("Enter comma separated resource names (PXI1Slot2,PXI1Slot3): ").split(",")
+
+# parameters
+sample_rate = 20000000.0
+waveform_size = 16
+# Create waveform data
+waveform_data = []
+for i in range(waveform_size):
+    waveform_data.append(0.0)
+waveform_data[0] = 1.0
+
+# Read in cmd args
+if len(sys.argv) >= 2:
+    server_address = sys.argv[1]
+if len(sys.argv) >= 3:
+    server_port = sys.argv[2]
+
+# Create the communication channel for the remote host and create connections to the NI-FGEN and session services.
+channel = grpc.insecure_channel(f"{server_address}:{server_port}")
+nifgen_service = grpc_nifgen.NiFgenStub(channel)
+nitclk_service = grpc_nitclk.NiTClkStub(channel)
+
+any_error = False
+# Checks for errors. If any, throws an exception to stop the execution.
+def CheckForError (service, vi, status) :
+    global any_error
+    if status != 0 and not any_error:
+        any_error = True
+        ThrowOnError (service, vi, status)
+
+# Converts an error code returned by NI-FGEN into a user-readable string.
+def ThrowOnError (service, vi, error_code):
+    if service == nifgen_service:
+        error_message_request = nifgen_types.ErrorMessageRequest(
+            vi = vi,
+            error_code = error_code
+        )
+        error_message_response = nifgen_service.ErrorMessage(error_message_request)
+        raise Exception (error_message_request.error_message)
+    else:
+        error_message_request = nitclk_types.GetExtendedErrorInfoRequest()
+        error_message_response = nitclk_service.GetExtendedErrorInfo(error_message_request)           
+        raise Exception (error_message_response.error_string)
+
+try:
+    # list of sessions
+    sessions = []
+    i = 0
+    for resource in resources:
+        # Initalize NI-FGEN session
+        init_with_channels_resp = nifgen_service.InitWithOptions(nifgen_types.InitWithOptionsRequest(
```

Comment:
Change variable name to match API you are calling.

Reply:
fixed